backend/corpus_asset_pipelines/summary_cubes/expression_summary/transform.py

```diff
@@ -52,47 +52,48 @@ def reduce_X(tdb_group: str, cube_indices, *accum):
             iterable = expression.query(return_incomplete=True, order="U", attrs=["rankit"])
             future = None
             for i, result in enumerate(iterable.df[:]):
-                # todo pull out into sep function
                 logger.info(f"reduce integrated expression data, iter {i}")
                 if future is not None:
                     future.result()  # forces a wait
-                future = tp.submit(
-                    coo_cube_pass1_into,
+                future = executor.submit(
+                    gene_expression_sum_x_cube_dimension,
                     result["rankit"].values,
                     result["obs_idx"].values,
                     result["var_idx"].values,
                     cube_indices,
-                    *accum,
+                    cube_sum,
+                    cube_nnz,
                 )
 
-        return accum
-
 
 # TODO: this could be further optimize by parallel chunking.  Might help large arrays if compute ends up being a bottleneck. # noqa E501
 @nb.njit(fastmath=True, error_model="numpy", parallel=False, nogil=True)
-def coo_cube_pass1_into(data, row, col, row_groups, sum_into, nnz_into, min_into, max_into):
+def gene_expression_sum_x_cube_dimension(
```

Comment:
Bless your heart for this rename!

Reply:
loll its still not perfect but I figure its waaaaaay better and someone else can build on it in the future